cmd/vsphere-xcopy-volume-populator/internal/pure/flashArray.go

```diff
@@ -0,0 +1,192 @@
+package pure
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/devans10/pugo/flasharray"
+	"github.com/kubev2v/forklift/cmd/vsphere-xcopy-volume-populator/internal/populator"
+	"k8s.io/klog/v2"
+)
+
+const FlashProviderID = "624a9370"
+
+type FlashArrayClonner struct {
+	client *flasharray.Client
+}
+
+func NewFlashArrayClonner(hostname, username, password string, skipSSLVerification bool) (FlashArrayClonner, error) {
+	client, err := flasharray.NewClient(
+		hostname, username, password, "", "", true, false, "", map[string]string{})
+	if err != nil {
+		return FlashArrayClonner{}, err
+	}
+	array, err := client.Array.Get(nil)
+	if err != nil {
+		klog.Fatalf("Error getting array status: %v", err)
+	}
+	klog.Infof("Array Name: %s, ID: %s", array.ArrayName, array.ID)
+	return FlashArrayClonner{client: client}, nil
+}
+
+// EnsureClonnerIgroup creates or updates an initiator group with the clonnerIqn
+func (f *FlashArrayClonner) EnsureClonnerIgroup(initiatorGroup string, clonnerIqn []string) (populator.MappingContext, error) {
+
+	if true {
+		// pure does allow a single host to connect to 2 separae groups. Hence
+		// we must connect map the volume to the host, and not to the group
+		hostNames := []string{}
+		hosts, err := f.client.Hosts.ListHosts(nil)
+		if err != nil {
+			return nil, err
+		}
+		for _, h := range hosts {
+			for _, iqn := range h.Iqn {
+				if slices.Contains(clonnerIqn, iqn) {
+					klog.Infof("adding host to group %v", h.Name)
+					hostNames = append(hostNames, h.Name)
+				}
+			}
+			for _, wwn := range h.Wwn {
+				if slices.Contains(clonnerIqn, wwn) {
+					klog.Infof("adding host to group %v", h.Name)
+					hostNames = append(hostNames, h.Name)
+				}
+			}
+		}
+		return populator.MappingContext{"hosts": hostNames}, nil
+	}
+
+	klog.Infof("starting ensure clonner group with %v %v", initiatorGroup, clonnerIqn)
+	g, err := f.client.Hostgroups.GetHostgroup(initiatorGroup, nil)
+	if err != nil {
```

Comment:
I see initiatorGroup is a hardcoded name. How do we handle cleanup from previous workloads, either successful or failed in the middle.
and if we have concurrent executions of this populator workload, wouldn't this run into problems when everybody is dealing with the same hgroup?